applications/main/nfc/plugins/supported_cards/ndef.c

```diff
@@ -86,29 +98,56 @@ static void parse_ndef_uri(FuriString* str, const uint8_t* payload, uint32_t pay
         prepend = prepends[prepend_type];
     }
     size_t prepend_len = strlen(prepend);
-
     size_t uri_len = prepend_len + (payload_len - 1);
-    char* const uri_buf = malloc(uri_len);
+
+    char* uri_buf = malloc(uri_len + 1);
+    if(!uri_buf) {
+        furi_string_cat_printf(str, "Memory allocation failed\n");
+        return;
+    }
+
     memcpy(uri_buf, prepend, prepend_len);
     memcpy(uri_buf + prepend_len, payload + 1, payload_len - 1);
-    char* uri = uri_buf;
+    uri_buf[uri_len] = '\0';
```

Comment:
and flipper malloc() also does memset(0) so its already \0 at the end

Reply:
not relevant anymore, i changed it to print exact number of chars, so dont even need a terminator, we know the length